service_test.go

```diff
@@ -262,6 +262,42 @@ func TestServer_SetupReadWriteTimeouts(t *testing.T) {
 	}
 }
 
+func TestServer_SetupDeflateLevel(t *testing.T) {
+	tests := []struct {
+		name      string
+		component Component
+		ctx       context.Context
+		level     string
+		wantErr   bool
+	}{
+		{name: "success without setup compression deflate level", component: &testComponent{}, ctx: context.Background(), wantErr: false},
+		{name: "success with setup compression deflate level = -2", component: &testComponent{}, ctx: context.Background(), level: "-2", wantErr: false},
+		{name: "success with setup compression deflate level = 2", component: &testComponent{}, ctx: context.Background(), level: "2", wantErr: false},
+		{name: "success with setup compression deflate level = 6", component: &testComponent{}, ctx: context.Background(), level: "6", wantErr: false},
+		{name: "success with setup compression deflate level = 9", component: &testComponent{}, ctx: context.Background(), level: "9", wantErr: false},
+		{name: "failed with too small compression deflate level", component: &testComponent{}, ctx: context.Background(), level: "-3", wantErr: true},
+		{name: "failed with too big compression deflate level", component: &testComponent{}, ctx: context.Background(), level: "10", wantErr: true},
+		{name: "failed with invalid compression deflate level", component: &testComponent{}, ctx: context.Background(), level: "blah", wantErr: true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.level != "" {
+				err := os.Setenv("PATRON_COMPRESSION_DEFLATE_LEVEL", tt.level)
+				assert.NoError(t, err)
+			}
+			svc, err := New("test", "", TextLogger())
+			require.NoError(t, err)
+
+			_, err = svc.WithComponents(tt.component, tt.component, tt.component).build()
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
```

Comment:
You might also want to set valid values for `PATRON_HTTP_READ_TIMEOUT` and `PATRON_HTTP_WRITE_TIMEOUT`, as the CI is failing.

Reply:
Ah, I see. Didn't have time on Friday to check why it's failing, thanks for pointing out

Reply:
Fixed https://github.com/beatlabs/patron/pull/306/commits/09c3584c09ad33d2efbe2b774e729cf3e7622515

I think instead of fixing the tests by setting unrelated values, it's better to clean up after tests and keep the tests related to a single problem they cover :)